crypto/sleep.c

```diff
@@ -44,13 +44,34 @@ void OSSL_sleep(uint64_t millis)
 
 void OSSL_sleep(uint64_t millis)
 {
-    Sleep(millis);
+    /*
+     * Windows' Sleep() takes a DWORD argument, which is smaller than
+     * a uint64_t, so we need to split the two to shut the compiler up.
+     */
+    DWORD dword_times = (DWORD)(millis >> sizeof(DWORD));
+    DWORD i;
+
+    for(i = dword_times; i-- > 0;)
+        Sleep((DWORD)-1);
+    /*
+     * The loop above slept 1 millisec less on each iteration than it should,
+     * this compensates by sleeping as many milliseconds as there were
+     * iterations.  Yes, this is nit picky!
+     */
+    Sleep(dword_times);
```

Comment:
Unless Windows' `Sleep(0);` is a no-op and not yielding CPU to other processes, this is wrong. Because we will otherwise yield CPU twice even for small `millis` values. This should be called only when `dword_times > 0`.

Reply:
Okie, fixed